lib/queue.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+
+var async = require('async')
+var chalk = require('chalk')
+var debug = require('debug-log')('mashape-analytics-queue')
+var http = require('http')
+var pkg = require('../package.json')
+var util = require('util')
+
+/**
+ * ALF Queue
+ */
+var Queue = module.exports = function (serviceToken, environment, options) {
+  var self = this
+
+  this.opts = options
+  this.timeout = null
+  this.userAgent = util.format('%s/%s', 'mashape-analytics-agent-node', pkg.version)
+
+  // setup event queue
+  // TODO specify worker pool
+  // TODO use msgpack + gzip?
+  this.queue = async.queue(function (entry, done) {
+    // append entry to log
+    self.alf.har.log.entries.push(entry)
+
+    // start timer
+    if (!self.timeout) {
+      self.timeout = setTimeout(function () {
```

Comment:
separation of concerns. the idea is the simplify the code, not add complexity.

expected input: HAR entries
expected output: Send to socket server

Simple.

You're suggesting:
expected input: HAR entries
expected controls: start flush timer
expected output: Send to socket server

And if you want to align the responsibility of flushing outside of the queue, you might as well also remove the check for full bucket size.

expected input: HAR entries
expected controls: start flush timer, check for full bucket
expected output: Send to socket server

Then you might as well not abstract queue out, and leave the queue as an array, and the logic in index, which means, no separation of concerns; you're left with all logic in one file again.